app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -50,162 +49,135 @@ class ThreadAdapter : RecyclerView.Adapter<BindingViewHolder<ItemMessageBinding>
 
     override fun getItemCount() = messageList.size
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BindingViewHolder<ItemMessageBinding> =
-        BindingViewHolder(ItemMessageBinding.inflate(LayoutInflater.from(parent.context), parent, false))
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BindingViewHolder<ItemMessageBinding> {
+        return BindingViewHolder(ItemMessageBinding.inflate(LayoutInflater.from(parent.context), parent, false))
+    }
 
-    override fun onBindViewHolder(holder: BindingViewHolder<ItemMessageBinding>, position: Int) {
+    override fun onBindViewHolder(holder: BindingViewHolder<ItemMessageBinding>, position: Int): Unit = with(holder.binding) {
         val message = messageList[position]
-        with(holder) {
-            displayHeader(binding, message)
-            displayAttachments(binding, message.attachments)
-            displayBody(binding, message.body)
-        }
+        displayHeader(message)
+        displayAttachments(message.attachments)
+        displayBody(message.body)
     }
 
-    private fun displayHeader(binding: ItemMessageBinding, message: Message) {
-        with(binding.headerThread) {
-            expeditorName.text = message.from[0].name.ifBlank { message.from[0].email }
-            expeditorEmail.text = message.from[0].email
-            recipient.text = formatRecipientsName(binding.root.context, message)
-            expandHeaderButton.setOnClickListener {
-                val isExpanded = !message.isExpandedHeaderMode
-                message.isExpandedHeaderMode = isExpanded
-                expandHeader(this, message)
-            }
-            messageDate.text = message.date?.toDate()?.format("d MMM YYYY à HH:mm")
+    fun notifyAdapter(newList: ArrayList<Message>) {
+        DiffUtil.calculateDiff(MessageListDiffCallback(messageList, newList)).dispatchUpdatesTo(this)
+        messageList = newList
+    }
+
+    private fun ItemMessageBinding.displayHeader(message: Message) {
+        expeditorName.text = message.from[0].name.ifBlank { message.from[0].email }
+        expeditorEmail.text = message.from[0].email
+        messageDate.text = message.date?.toDate()?.format("d MMM YYYY à HH:mm")
```

Comment:
Use `FORMAT_DATE_CLEAR_MONTH` + `FORMAT_DATE_HOUR_MINUTE` and handle this hardcoded `à`.

Reply:
Done.